app/src/main/kotlin/com/wire/android/datastore/GlobalDataStore.kt

```diff
@@ -141,47 +164,93 @@ class GlobalDataStore @Inject constructor(@ApplicationContext private val contex
 
-    // returns a flow with decoded passcode
+    /**
+     * returns a flow with decoded passcode
+     */
     @Suppress("TooGenericExceptionCaught")
-    fun getAppLockPasscodeFlow(): Flow<String?> =
-        context.dataStore.data.map {
-            it[APP_LOCK_PASSCODE]?.let {
+    fun getAppLockPasscodeFlow(): Flow<String?> {
+        val preference = if (isAppLockPasscodeSet()) APP_LOCK_PASSCODE else TEAM_APP_LOCK_PASSCODE
+        return context.dataStore.data.map {
+            it[preference]?.let { passcode ->
                 try {
-                    EncryptionManager.decrypt(APP_LOCK_PASSCODE.name, it)
+                    EncryptionManager.decrypt(preference.name, passcode)
                 } catch (e: Exception) {
                     null
                 }
             }
         }
+    }
 
-    // returns a flow only informing whether the passcode is set, without the need to decode it
+    /**
+     * returns a flow only informing whether the passcode is set, without the need to decode it
+     */
     fun isAppLockPasscodeSetFlow(): Flow<Boolean> =
         context.dataStore.data.map {
             it.contains(APP_LOCK_PASSCODE)
         }
 
+    fun isAppLockPasscodeSet(): Boolean = runBlocking {
+        context.dataStore.data.map {
+            it.contains(APP_LOCK_PASSCODE)
+        }.first()
+    }
+
+    fun isAppTeamPasscodeSet(): Boolean = runBlocking {
+        context.dataStore.data.map {
+            it.contains(TEAM_APP_LOCK_PASSCODE)
+        }.first()
+    }
+
     suspend fun clearAppLockPasscode() {
         context.dataStore.edit {
             it.remove(APP_LOCK_PASSCODE)
         }
     }
 
+    suspend fun clearTeamAppLockPasscode() {
+        context.dataStore.edit {
+            it.remove(TEAM_APP_LOCK_PASSCODE)
+        }
+    }
+
     @Suppress("TooGenericExceptionCaught")
-    suspend fun setAppLockPasscode(passcode: String) {
+    private suspend fun setAppLockPasscode(
+        passcode: String,
+        key: Preferences.Key<String>
+    ) {
         context.dataStore.edit {
             try {
-                val encrypted = EncryptionManager.encrypt(APP_LOCK_PASSCODE.name, passcode)
-                it[APP_LOCK_PASSCODE] = encrypted
+                val encrypted =
+                    EncryptionManager.encrypt(key.name, passcode)
+                it[key] = encrypted
             } catch (e: Exception) {
-                it.remove(APP_LOCK_PASSCODE)
+                it.remove(key)
             }
         }
     }
 
+    suspend fun setTeamAppLock(
+        passcode: String,
+        key: Preferences.Key<String> = TEAM_APP_LOCK_PASSCODE
+    ) {
+        setAppLockPasscode(passcode, key)
+    }
+
+    suspend fun setUserAppLock(
+        passcode: String,
+        key: Preferences.Key<String> = APP_LOCK_PASSCODE
+    ) {
+        setAppLockPasscode(passcode, key)
+    }
+
+    fun isAppLockedByUserFlow(): Flow<Boolean> =
+        getBooleanPreference(IS_APP_LOCKED_BY_USER, false)
```

Comment:
question: this value with key `IS_APP_LOCKED_BY_USER` has only getter, but no setter, so it cannot be changed - will that be added in a following PR?

Reply:
Yeah it's left over and it's not used, removing it